core/domain/exp_services_test.py

```diff
@@ -5189,6 +5528,121 @@ def test_get_all_exploration_summaries(self):
+        rights_manager.publish_exploration(self.owner, self.EXP_0_ID)
+        rights_manager.publish_exploration(self.owner, self.EXP_1_ID)
+        rights_manager.publish_exploration(self.owner, self.EXP_2_ID)
+        exploration_summaries = (
+            exp_services.get_recently_published_exp_summaries(3)
+        )
+        recently_published_summaries = (
+            exp_models.ExpSummaryModel.get_recently_published(3))
+        recently_publshed_summaries_model = (
+            exp_fetchers.get_exploration_summaries_from_models(
+                recently_published_summaries))
+        self.assertEqual(len(exploration_summaries), 3)
+        self.assertItemsEqual(
+        exploration_summaries,
+        recently_publshed_summaries_model)
+
+    def test_get_story_id_linked_to_exploration(self):
+        self.assertIsNone(
+            exp_services.get_story_id_linked_to_exploration(self.EXP_ID_1))
+        story_id = story_services.get_new_story_id()
+        topic_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id, self.albert_id, name='Topic',
+            abbreviated_name='topic-one', url_fragment='topic-one',
+            description='A new topic',
+            canonical_story_ids=[], additional_story_ids=[],
+            uncategorized_skill_ids=['skill_4'], subtopics=[],
+            next_subtopic_id=0)
+        self.save_new_story(story_id, self.albert_id, topic_id)
+        topic_services.add_canonical_story(self.albert_id, topic_id, story_id)
+        change_list = [
+            story_domain.StoryChange({
+                'cmd': story_domain.CMD_ADD_STORY_NODE,
+                'node_id': story_domain.NODE_ID_PREFIX + '1',
+                'title': 'Title 1'
+            }),
+             story_domain.StoryChange({
+                'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY,
+                'property_name': (
+                    story_domain.STORY_NODE_PROPERTY_EXPLORATION_ID),
+                'node_id': story_domain.NODE_ID_PREFIX + '1',
+                'old_value': None,
+                'new_value': self.EXP_ID_1
+            })
+        ]
+        story_services.update_story(
+            self.albert_id, story_id, change_list,
+            'Added node.')
+        self.assertEqual(
+            exp_services.get_story_id_linked_to_exploration(self.EXP_ID_1),
+            story_id)
+
+    def test_get_user_exploration_data(self):
+        self.save_new_valid_exploration(self.EXP_0_ID, self.albert_id)
+        exploration_description = (
+            exp_services.get_user_exploration_data(
+                self.albert_id, self.EXP_0_ID))
+        self.assertIsNotNone(exploration_description)
+
+        exploration = self.save_new_valid_exploration(
+            self.EXP_0_ID,
+            self.albert_id)
+        exp_services._save_exploration(self.albert_id, exploration, '', [])  # pylint: disable=protected-access
```

Comment:
Why is this accessing of a private method necessary? The end user can save and retrieve exploration data using public methods, so shouldn't the test be able to do so also?